src/unit-number/unit-number-input.tsx

```diff
@@ -188,69 +188,70 @@ export const UnitNumberInput = ({
             : value.toLocaleUpperCase();
     };
 
-    const formatDisplayValues = (value: string | undefined) => {
-        if (value === undefined || value === "") {
-            setFloorValue("");
-            setUnitValue("");
-        } else {
+    const updateValues = (value: string | undefined) => {
+        if (value === INVALID_VALUE) {
+            return;
+        }
+
+        if (!(value === undefined || value.length === 0)) {
             const valueArr = value.split("-");
             if (valueArr.length !== 0) {
                 // Valid value
                 const floor = valueArr[0];
                 const unit = valueArr[1];
-
-                setFloorValue(formatInput(floor));
-                setUnitValue(formatInput(unit));
+                setFloorValue(
+                    currentFocus === "floor" ? floor : formatInput(floor)
+                );
+                setUnitValue(
+                    currentFocus === "unit" ? unit : formatInput(unit)
+                );
             }
+        } else {
+            setFloorValue("");
+            setUnitValue("");
         }
     };
 
     const performOnChangeHandler = (changeValue: string, field: FieldType) => {
-        if (onChange) {
-            const values: Record<ValueFieldTypes, string> = {
-                floor: floorValue,
-                unit: unitValue,
-            };
+        if (!onChange && !onChangeRaw) {
+            return;
+        }
 
-            // Update the specific field value
-            values[field] = changeValue;
+        const values: Record<ValueFieldTypes, string> = {
+            floor: floorValueStateRef.current,
+            unit: unitValueStateRef.current,
+        };
 
+        // Update the specific field value
+        values[field] = changeValue;
+
+        if (onChange) {
             const returnValue = getFormattedValue(values);
             onChange(returnValue);
         }
 
         if (onChangeRaw) {
-            const valuesArr = [
-                ...(field === "floor"
-                    ? [formatInput(changeValue)]
-                    : [floorValue]),
-                ...(field === "unit"
-                    ? [formatInput(changeValue)]
-                    : [unitValue]),
-            ];
-
-            onChangeRaw(valuesArr);
+            onChangeRaw([values.floor, values.unit]);
```

Comment:
We'll not format the values onChange unless the input has been blurred

Reply:
just want to clarify: does input here mean the individual floor/unit input elements? not the entire component?

so this would be the expected behaviour?
1. type 1 in `floor` -> `onChange("Invalid Unit Number")`
2. switch to unit input -> floor gets padded visually, but no change/blur handlers triggered
3. type 1 in `unit` -> `onChange("01-1")`

Reply:
Yes the input I am referring to is the entire component. 

And yes the behaviour you listed is expected.